Source/Core/Core/Slippi/SlippiUser.cpp

```diff
@@ -132,6 +134,202 @@ SlippiUser::~SlippiUser()
+		return RANK_DIAMOND_3;
+	if (ratingOrdinal >= 2191.75 && globalPlacing && regionalPlacing)
+		return RANK_GRANDMASTER;
+	if (ratingOrdinal > 2191.75 && ratingOrdinal <= 2274.99)
+		return RANK_MASTER_1;
+	if (ratingOrdinal > 2275 && ratingOrdinal <= 2350)
+		return RANK_MASTER_2;
+	if (ratingOrdinal > 2350)
+		return RANK_MASTER_3;
+	return RANK_UNRANKED;
+}
+
+SlippiUser::RankInfo SlippiUser::FetchUserRank(std::string connectCode)
+{
+	RankInfo info;
+
+	const char *query = 
+		"fragment profileFields on NetplayProfile {\n"
+		"  id\n"
+		"  ratingOrdinal\n"
+		"  ratingUpdateCount\n"
+		"  wins\n"
+		"  losses\n"
+		"  dailyGlobalPlacement\n"
+		"  dailyRegionalPlacement\n"
+		"  continent\n  characters {\n"
+		"    id\n"
+		"    character\n"
+		"    gameCount\n"
+		"    __typename\n"
+		"  }\n"
+		"  __typename\n"
+		"}\n"
+		"\n"
+		"fragment userProfilePage on User {\n"
+		"  fbUid\n"
+		"  displayName\n"
+		"  connectCode {\n"
+		"    code\n"
+		"    __typename\n"
+		"  }\n"
+		"  status\n"
+		"  activeSubscription {\n"
+		"    level\n"
+		"    hasGiftSub\n"
+		"    __typename\n"
+		"  }\n"
+		"  rankedNetplayProfile {\n"
+		"    ...profileFields\n"
+		"    __typename\n"
+		"  }\n"
+		"  netplayProfiles {\n"
+		"    ...profileFields\n"
+		"    season {\n"
+		"      id\n"
+		"      startedAt\n"
+		"      endedAt\n"
+		"      name\n"
+		"      status\n"
+		"      __typename\n"
+		"    }\n"
+		"    __typename\n"
+		"  }\n"
+		"  __typename\n"
+		"}\n"
+		"\n"
+		"query AccountManagementPageQuery($cc: String!, $uid: String!) {\n"
+		"  getUser(fbUid: $uid) {\n"
+		"    ...userProfilePage\n"
+		"    __typename\n"
+		"  }\n"
+		"  getConnectCode(code: $cc) {\n"
+		"    user {\n"
+		"      ...userProfilePage\n"
+		"      __typename\n"
+		"    }\n"
+		"    __typename\n"
+		"  }\n"
+		"}\n";
+
+	std::string url = "https://gql-gateway-dot-slippi.uc.r.appspot.com/graphql";
+	json body = {{"operationName", "AccountManagementPageQuery"},
+		{"variables", {{"cc", connectCode}, {"uid", connectCode}}},
+		{"query", query}};
+
+	// INFO_LOG(SLIPPI_ONLINE, "Preparing request...");
+	// Perform curl request
+	std::string resp;
+	curl_easy_setopt(m_curl, CURLOPT_URL, (url).c_str());
+	curl_easy_setopt(m_curl, CURLOPT_WRITEDATA, &resp);
+	curl_easy_setopt(m_curl, CURLOPT_POSTFIELDS, (body.dump()).c_str());
+	curl_easy_setopt(m_curl, CURLOPT_CUSTOMREQUEST, "POST");
+
+	CURLcode res = curl_easy_perform(m_curl);
+
+	// INFO_LOG(SLIPPI_ONLINE, "Request sent");
+	if (res != 0)
+	{
+	    ERROR_LOG(SLIPPI, "[User] Error fetching user info from server, code: %d", res);
+	    //return info;
+	}
+
+	long responseCode;
+	curl_easy_getinfo(m_curl, CURLINFO_RESPONSE_CODE, &responseCode);
+	if (responseCode != 200)
+	{
+	    ERROR_LOG(SLIPPI, "[User] Server responded with non-success status: %d", responseCode);
+	    //return info;
+	}
+
+	auto r = json::parse(resp);
+	auto rankedObject = r["data"]["getConnectCode"]["user"]["rankedNetplayProfile"];
```

Comment:
should probably make sure `json::parse(resp)` returned something sane before you start dereferencing everything.